lib/ecto/adapters/mysql/connection.ex

```diff
@@ -514,7 +514,7 @@ if Code.ensure_loaded?(Mariaex.Connection) do
 
     defp ecto_to_db(:string),     do: "varchar"
     defp ecto_to_db(:float),      do: "double"
-    defp ecto_to_db(:datetime),   do: "datetime"
+    defp ecto_to_db(:datetime),   do: "datetime(6)"
```

Comment:
People don't upgrade databases as often as everything else as it requires replication and complicated setup if you want zero downtime updates and we need to run on legacy databases just as well. Given we were considering at least SQL Server 2008 for TDS, MySQL 5.6 is too recent to restrict only to it.